node/src/actors/chain_manager/mod.rs

```diff
@@ -278,54 +271,55 @@ impl ChainManager {
                     checkpoint: block_epoch,
                     hash_prev_block: block_hash,
                 };
+
                 chain_info.highest_block_checkpoint = beacon;
+                update_pools(
+                    &block,
+                    &mut self.chain_state.unspent_outputs_pool,
+                    &mut self.chain_state.data_request_pool,
+                    &mut self.transactions_pool,
+                    utxo_diff,
+                );
 
-                // Update UnspentOutputsPool
-                self.chain_state.unspent_outputs_pool = utxo_set;
-
-                // Update TransactionPool
-                update_transaction_pool(&mut self.transactions_pool, block.txns.as_ref());
-
-                // Update DataRequestPool
-                self.chain_state.data_request_pool = dr_pool;
-                let reveals = self
-                    .chain_state
-                    .data_request_pool
-                    .update_data_request_stages();
-                for reveal in reveals {
-                    // Send AddTransaction message to self
-                    // And broadcast it to all of peers
-                    ctx.address().do_send(AddTransaction {
-                        transaction: reveal,
-                    })
-                }
-                // Persist finished data requests into storage
-                let to_be_stored = self.chain_state.data_request_pool.finished_data_requests();
-                to_be_stored.into_iter().for_each(|dr| {
-                    self.persist_data_request(ctx, &dr);
-                    if info_flag {
+                // Insert candidate block into `block_chain` and persist it
+                self.chain_state.block_chain.insert(block_epoch, block_hash);
+
+                // Persist chain_info into storage
```

Comment:
Move this comment to its properly place, near to `self.persist_chain_state(ctx);` in line 316